docs(sql): adding hint of injection token with typeorm async Connection options (#1577)
nestjs/docs.nestjs.com: content/techniques/sql.md

> warning **Notice** If you don't set the `name` for a connection, its name is set to `default`. Please note that you shouldn't have multiple connections without a name, or with the same name, otherwise they will get overridden.

> If you using multiple connections with asynchronously, set Connection Name like this.

Reviewer: ```suggestion
If you need to use multiple connections with asynchronous module registration, you can set the name for the connection like this:
```

This construction works the same as `useClass` with one critical difference - `TypeOrmModule` will lookup imported modules to reuse an existing `ConfigService` instead of instantiating a new one.

If you don't use ConnectionName as 'default', you need to specify ConnectionName so that nestjs can recognize it.

> > **Hint**: make sure the `name` property is on the same level as the `useFactory`, `useClass`, or `useValue` property. This will allow Nest to properly create the injection token

Reviewer: ```suggestion

> info **Hint**: make sure the `name` property is on the same level as the `useFactory`, `useClass`, or `useValue` property. This will allow Nest to properly create the injection token
```

Sorry, had the wrong syntax. This is the last edit, I promise. This will make the hint show up in blue like it does on Line #645

This construction works the same as `useClass` with one critical difference - `TypeOrmModule` will lookup imported modules to reuse an existing `ConfigService` instead of instantiating a new one.

> If you don't use ConnectionName as 'default', you need to specify ConnectionName so that nestjs can recognize it.
> 
> > info **Hint**: make sure the `name` property is on the same level as the `useFactory`, `useClass`, or `useValue` property. This will allow Nest to properly create the injection token
> 
> 

Reviewer 2: ```suggestion
> info **Hint** Make sure that the `name` property is defined on the same level as the `useFactory`, `useClass`, or `useValue` property. This will allow Nest to properly register the connection under the appropriate injection token.

```

This construction works the same as `useClass` with one critical difference - `TypeOrmModule` will lookup imported modules to reuse an existing `ConfigService` instead of instantiating a new one.

If you don't use ConnectionName as 'default', you need to specify ConnectionName so that nestjs can recognize it.

> 
> ```typescript
> TypeOrmModule.forRootAsync({
>   useFactory: () => {
>     return {
>       type: 'mysql',
>       host: 'localhost',
>       port: 3306,
>       username: 'root',
>       password: 'root',
>       database: 'test',
>       entities:[],
>       synchronize: true,
>       name:"typeorm_user_connection"
>     }
>   },
>   name: 'USER',
> }),
> ```

Reviewer: ```suggestion
> **Hint**: make sure the `name` property is on the same level as the `useFactory`, `useClass`, or `useValue` property. This will allow Nest to properly create the injection token
```

This will help allow cut down on code, and still illustrate the point being made, in my opinion

Author: ok, i am satisfied.

This construction works the same as `useClass` with one critical difference - `TypeOrmModule` will lookup imported modules to reuse an existing `ConfigService` instead of instantiating a new one.

> > info **Hint** Make sure that the `name` property is defined on the same level as the `useFactory`, `useClass`, or `useValue` property. This will allow Nest to properly register the connection under the appropriate injection token.

Reviewer 2: ```suggestion
> info **Hint** Make sure that the `name` property is defined at the same level as the `useFactory`, `useClass`, or `useValue` property. This will allow Nest to properly register the connection under the appropriate injection token.
```
